src/web/components/ClientSideTokenGeneration/CstgAddDomainDialog.tsx

```diff
@@ -0,0 +1,86 @@
+import { useEffect, useState } from 'react';
+import { FormProvider, useForm } from 'react-hook-form';
+import { parse } from 'tldts';
+
+import { AddDomainNamesFormProps } from '../../services/domainNamesService';
+import { extractTopLevelDomain, separateStringsCommaSeparatedList } from '../../utils/textHelpers';
+import { Dialog } from '../Core/Dialog';
+import { InlineMessage } from '../Core/InlineMessage';
+import { TextInput } from '../Input/TextInput';
+
+import '../KeyPairs/KeyPairDialog.scss';
+
+type AddDomainNamesDialogProps = Readonly<{
+  onAddDomains: (newDomainNamesFormatted: string[]) => Promise<void>;
+  onOpenChange: () => void;
+}>;
+
+type DomainProps = {
+  isIcann: boolean | null;
+  isPrivate: boolean | null;
+  domain: string | null;
+};
+
+type ValidDomainProps = Omit<DomainProps, 'domain'> & { domain: string };
```

Comment:
This seems a little awkward.  omitting the prop and re-adding a slightly different one with the same name.  Also, can the fields on DomainProps just be optional?

Reply:
Bad example of me copying an existing example :/ ValidDomainProps isn't needed, can just have DomainProps and domain is oprtional